staging/src/k8s.io/apiserver/pkg/storage/etcd3/compact.go

```diff
@@ -121,7 +121,7 @@ func compactor(ctx context.Context, client *clientv3.Client, interval time.Durat
 
 		compactTime, rev, err = compact(ctx, client, compactTime, rev)
 		if err != nil {
-			klog.Errorf("etcd: endpoint (%v) compact failed: %v", client.Endpoints(), err)
+			klog.ErrorS(err, "etcd: endpoint (%v) compact failed", "endpoints", client.Endpoints())
```

Comment:
The remaining %v does not look correct if neither the error or the endpoints are an arg to the format string any more

Reply:
Do we have any kind of static code analysis that ensures a valid format string and number of args are passed to the structured logging methods (similar to the lint checks on fmt format strings performed by go lint / go test)?